### ISSUE TITLE
Bug: (fill in)

### ISSUE THREAD

**Issue by sstchur** [eslint/eslint#16184]
### Environment

Node version: 16.4.2
npm version: 8.5.0
Local ESLint version: 8.19.0
Global ESLint version: N/A
Operating System: Github Codespaces (Linux VM), but this repos on Windows and Linux and everywhere I've tried it

### What parser are you using?

@typescript-eslint/parser

### What did you do?

The simplest way to see the issue is to clone this repo: https://github.com/sstchur/eslint-repro

Run:
`npm install`
`node no-unstrict-equals.test.js`

You will see this error output:
```
Configuration for rule "no-unnecessary-explicit-falsy-checks" is invalid:
Value ["===","!=="] should NOT have more than 0 items.
```

**My tsconfig.json**
```json
nes (22 sloc)  602 Bytes

{
    "compilerOptions": {
      "target": "es2018",
      "module": "esnext",
      "jsx": "react",
      "declaration": true,
      "declarationMap": true,
      "sourceMap": true,
      "experimentalDecorators": true,
      "noEmitOnError": true,
      "moduleResolution": "node",
      "strict": true,
      "noFallthroughCasesInSwitch": true,
      "noImplicitReturns": true,
      "noUnusedLocals": true,
      "noUnusedParameters": true,
      "skipLibCheck": true,
      "forceConsistentCasingInFileNames": true,
      "importHelpers": true
    },
    "include": ["./**/*.js", "file.ts"]
  }
  ```

**my test file**
```typescript
const { RuleTester } = require('eslint');
const { ESLintUtils } = require('@typescript-eslint/utils');
const noUnstrictEquals = require('./no-unstrict-equals');

const ruleTester = new ESLintUtils.RuleTester(
{
  parser: '@typescript-eslint/parser',
  parserOptions: {
    project: './tsconfig.json',
    tsconfigRootDir: __dirname
  }
});

const errors = [{ messageId: 'noUnstrictEqauals' }];
ruleTester.run('no-unnecessary-explicit-falsy-checks', noUnstrictEquals, {
  valid: [
    {
      options: [ "===", "!==" ],
      code: `let n: number;
            if (n === 7) {};`
    }
  ],
  invalid: [
    {
      code: `let x: boolean = false;
             if (x == '') {}`,
      errors
    }
  ]
});
```

### What did you expect to happen?

I expect to be able to access `context.options` in my rule and have the rule behavior differently based on passed in options.  But I obviously need a way to write tests and specify options for each test as well.  That is the part that seems not to work.

I have search both Bing and Google for this error.  The only issues I found where people claiming this was caused by a global install of eslint, or an old version of eslint.  Neither of those situations applies here.

### What actually happened?

I get this error when trying to run my test:

```
Configuration for rule "no-unnecessary-explicit-falsy-checks" is invalid:
Value ["===","!=="] should NOT have more than 0 items.
```

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by sstchur** [comment]
I figured out the issue.  It was due to the absent-mindedly pasted `schema: []` which I totally didn't notice.  Closing this out.
